apidef/validator.go

```diff
@@ -137,3 +138,40 @@ func shouldValidateAuthSource(authType string, apiDef *APIDefinition) bool {
 
 	return false
 }
+
+var ErrInvalidIPCIDR = "invalid IP/CIDR %s"
+
+type RuleValidateIPList struct{}
+
+func (r *RuleValidateIPList) Validate(apiDef *APIDefinition, validationResult *ValidationResult) {
+	if apiDef.EnableIpWhiteListing {
+		if errs := r.validateIPAddr(apiDef.AllowedIPs); len(errs) > 0 {
+			validationResult.IsValid = false
+			validationResult.Errors = errs
+		}
+	}
+
+	if apiDef.EnableIpBlacklisting {
+		if errs := r.validateIPAddr(apiDef.BlacklistedIPs); len(errs) > 0 {
+			validationResult.IsValid = false
+			validationResult.Errors = append(validationResult.Errors, errs...)
+		}
+	}
+}
+
+func (r *RuleValidateIPList) validateIPAddr(ips []string) []error {
+	var errs []error
+	for _, ip := range ips {
+		_, _, err := net.ParseCIDR(ip)
```

Comment:
This could stand a `strings.Count` check for `/` before even attempting to parse CIDR; the loop should invoke ParseCIDR only if we reasonably asume that it's a CIDR.

```
if strings.Count(ip, "/") == 1 {
   // ParseCIDR
   continue
}

// ParseIP
```

Do we care about handling ParseCIDR errors? We don't log or return it.